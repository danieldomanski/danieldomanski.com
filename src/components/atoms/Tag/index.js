import React from 'react'
import { Link } from 'gatsby'
import styled, { withTheme } from 'styled-components'
import tw from 'tailwind.macro'

const Tag = styled(Link)`
  ${tw`font-sans text-xs px-4 py-2 text-primary-100 font-bold mr-2 rounded cursor-pointer no-underline bg-accent-700 hover:bg-accent-900 shadow`}
  transition: 0.1s ease-in-out;

  &:hover {
    background-color: ${props =>
      props.active ? props.activeHoverColor : props.hoverColor};
  }
`

const Filter = ({ data }) => {
  // ToDo kebabed case
  const { name, slug } = data

  return <Tag to={`/tags/${slug}`}>{name}</Tag>
}

export default withTheme(Filter)
