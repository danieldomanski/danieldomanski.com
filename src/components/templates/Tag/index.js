import React from 'react'
import styled from 'styled-components'
import { graphql } from 'gatsby'
import Title from '../../atoms/Text/Title'
import Box from '../../atoms/Box'
import Footer from '../../organisms/Footer'
import { usePageContent } from '../../../context/ContentContext'

const BlogPosts = styled.ul`
  margin: 4rem 0;
  list-style: none;
`

const Tag = ({ data, pageContext }) => {
  if (typeof window !== `undefined`) {
    usePageContent(data)
    const posts = data.posts.edges
    const { tag } = pageContext

    return (
      <>
        <Box m="auto" px={[6, 6, 12, 16, 24]} pb={8} flexGrow={1}>
          <Title>Posts about {tag}</Title>
          <BlogPosts>
            <PostsContainer posts={posts} />
          </BlogPosts>
        </Box>
        <Box as="footer" width={1} m="auto">
          <Footer variant="secondary" />
        </Box>
      </>
    )
  }

  return null
}

export default Tag

export const pageQuery = graphql`
  query PostsByTag($uid: String!, $locale: String!) {
    posts: allPrismicPost(
      filter: { data: { tags: { elemMatch: { tag: { uid: { eq: $uid } } } } } }
    ) {
      edges {
        node {
          uid
          first_publication_date
          last_publication_date
          data {
            title {
              text
            }
            tags {
              tag {
                document {
                  href
                  slugs
                  data {
                    tag
                  }
                }
                slug
              }
            }
            description {
              text
            }
            icon {
              text
            }
          }
        }
      }
    }
    footer: allPrismicFooter(filter: { lang: { eq: $locale } }) {
      edges {
        node {
          data {
            code_availability {
              text
            }
          }
        }
      }
    }
    header: allPrismicHeader(filter: { lang: { eq: $locale } }) {
      edges {
        node {
          data {
            brand {
              text
            }
            nav_home {
              text
            }
            nav_projects {
              text
            }
            nav_about {
              text
            }
            nav_articles {
              text
            }
          }
        }
      }
    }
  }
`
