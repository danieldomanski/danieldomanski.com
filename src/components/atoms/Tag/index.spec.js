import 'jest-styled-components'
import '@testing-library/jest-dom/extend-expect'
import React from 'react'
import { render, cleanup } from '@testing-library/react'

import Tag from './index'

afterEach(cleanup)

const mockData = {
  data: { name: 'test', slug: 'js' },
}

test('<Tag> renders with different styled-system category props', () => {
  const { container } = render(<Tag {...mockData}>Test</Tag>)

  expect(container.firstChild).toMatchSnapshot()
})
