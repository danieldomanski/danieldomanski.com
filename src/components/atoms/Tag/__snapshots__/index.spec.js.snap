// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tag> renders with different styled-system category props 1`] = `
.c0 {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  margin-right: 8px;
  color: primary.11;
  background-color: primary.1;
  display: inline-block;
  font-family: sans;
  font-size: xs;
  font-weight: medium;
}

<a
  bg="primary.1"
  class="c0"
  color="primary.11"
  display="inline-block"
  font-family="sans"
  font-size="xs"
  font-weight="medium"
  href="/tags/js"
  mr="2"
  px="3"
  py="2"
>
  test
</a>
`;
